pkg/agent/controller/supportbundlecollection/support_bundle_controller.go

```diff
@@ -0,0 +1,417 @@
+
+	UPLOAD_TO_FILESERVER_TRIES       = 5
+	UPLOAD_TO_FILESERVER_RETRY_DELAY = 5 * time.Second
+)
+
+var (
+	emptyWatch      = watch.NewEmptyWatch()
+	defaultFS       = afero.NewOsFs()
+	defaultExecutor = exec.New()
+	// Declared as variable for testing.
+	newAgentDumper = support.NewAgentDumper
+
+	getUploader          = getUploaderByProtocol
+	sftpUploaderInstance *sftpUploader
+)
+
+type SupportBundleController struct {
+	nodeName           string
+	nodeType           config.NodeType
+	namespace          string
+	antreaClientGetter agent.AntreaClientProvider
+	queue              *workqueue.Type
+	ovsCtlClient       ovsctl.OVSCtlClient
+	aq                 agentquerier.AgentQuerier
+	npq                querier.AgentNetworkPolicyInfoQuerier
+	v4Enabled          bool
+	v6Enabled          bool
+	lastCondition      v1alpha1.SupportBundleCollectionConditionType
+	conditionLock      sync.RWMutex
+}
+
+func NewSupportBundleController(nodeName string, nodeType config.NodeType, namespace string, antreaClientGetter agent.AntreaClientProvider, ovsCtlClient ovsctl.OVSCtlClient,
+	aq agentquerier.AgentQuerier, npq querier.AgentNetworkPolicyInfoQuerier, v4Enabled, v6Enabled bool) *SupportBundleController {
+	c := &SupportBundleController{
+		nodeName:           nodeName,
+		nodeType:           nodeType,
+		antreaClientGetter: antreaClientGetter,
+		queue:              workqueue.NewNamed("supportbundle"),
+		ovsCtlClient:       ovsCtlClient,
+		aq:                 aq,
+		npq:                npq,
+		v4Enabled:          v4Enabled,
+		v6Enabled:          v6Enabled,
+	}
+	return c
+}
+func (c *SupportBundleController) watchSupportBundleCollections() {
+	klog.Info("Starting watch for SupportBundleCollections")
+	antreaClient, err := c.antreaClientGetter.GetAntreaClient()
+	if err != nil {
+		klog.Warningf("Failed to get antrea client: %v", err)
+		return
+	}
+	options := metav1.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("nodeName", c.nodeName).String(),
+	}
+	watcher, err := antreaClient.ControlplaneV1beta2().SupportBundleCollections().Watch(context.TODO(), options)
+	if err != nil {
+		klog.Warningf("Failed to start watch for SupportBundleCollections: %v", err)
+		return
+	}
+	// Watch method doesn't return error but "emptyWatch" in case of some partial data errors,
+	// e.g. timeout error. Make sure that watcher is not empty and log warning otherwise.
+	if reflect.TypeOf(watcher) == reflect.TypeOf(emptyWatch) {
+		klog.Warning("Failed to start watch for SupportBundleCollections, please ensure antrea service is reachable for the agent")
+		return
+	}
+
+	klog.Info("Started watch for SupportBundleCollections")
+	eventCount := 0
+	defer func() {
+		klog.InfoS("Stopped watch for SupportBundleCollections", "total items received", eventCount)
+		watcher.Stop()
+	}()
+
+	for {
+		select {
+		case event, ok := <-watcher.ResultChan():
+			if !ok {
+				return
+			}
+			switch event.Type {
+			case watch.Bookmark:
+				klog.Info("Received Bookmark event")
```

Comment:
Does it need to be logged at default level? It doesn't seem useful in production.
I found all logging added by this PR is default level, could you adjust them based on their importance and frequency? A reference: https://github.com/tnqn/code-review-comments#do-not-make-all-logs-default-level